services/wallet/api.go

```diff
@@ -176,14 +176,14 @@ func (api *API) AddSavedAddress(ctx context.Context, sa SavedAddress) error {
 	if sa.ChainID == 0 {
 		sa.ChainID = api.s.rpcClient.UpstreamChainID
 	}
-	err := api.s.savedAddressesManager.AddSavedAddress(sa)
+	_, err := api.s.savedAddressesManager.AddOrUpdateSavedAddress(sa)
```

Comment:
`UpsertSavedAddress` maybe? a bit less wordy

Reply:
Will do.

P.S. I personally prefer a more human-readable approach to saving a few letters.

Reply:
Me too :), I find Upsert more readable since I am more familiar with it as commonly used for API/db operations, but either are fine by me